cmd/kubelet/kubelet.go

```diff
@@ -21,108 +21,31 @@ limitations under the License.
 package main
 
 import (
-	"fmt"
+	goflag "flag"
 	"os"
 
-	"github.com/golang/glog"
 	"github.com/spf13/pflag"
 
-	utilfeature "k8s.io/apiserver/pkg/util/feature"
-	"k8s.io/apiserver/pkg/util/flag"
+	utilflag "k8s.io/apiserver/pkg/util/flag"
 	"k8s.io/apiserver/pkg/util/logs"
 	"k8s.io/kubernetes/cmd/kubelet/app"
-	"k8s.io/kubernetes/cmd/kubelet/app/options"
 	_ "k8s.io/kubernetes/pkg/client/metrics/prometheus" // for client metric registration
 	_ "k8s.io/kubernetes/pkg/version/prometheus"        // for version metric registration
-	"k8s.io/kubernetes/pkg/version/verflag"
 )
 
-func parseFlagSet(fs *pflag.FlagSet, args []string) error {
-	if err := fs.Parse(args); err != nil {
-		return err
-	}
-	fs.VisitAll(func(flag *pflag.Flag) {
-		glog.V(2).Infof("FLAG: --%s=%q", flag.Name, flag.Value)
-	})
-	return nil
-}
-
-func die(err error) {
-	fmt.Fprintf(os.Stderr, "error: %v\n", err)
-	os.Exit(1)
-}
-
 func main() {
-	fs := pflag.NewFlagSet(os.Args[0], pflag.ExitOnError)
-	// set the normalize func, similar to k8s.io/apiserver/pkg/util/flag/flags.go:InitFlags
-	fs.SetNormalizeFunc(flag.WordSepNormalizeFunc)
-	// explicitly add flags from libs that register global flags
-	options.AddGlobalFlags(fs)
```

Comment:
does the switch to cobra cmd expose us to picking up all the globally registered flags again?

Reply:
> does the switch to cobra cmd expose us to picking up all the globally registered flags again?

Nothing new that I know.  It makes some of the previously registered ones more obvious via help since the help is accurate now, but it doesn't mess with the parsing.

Reply:
this was selectively adding specific global flags, not all global flags, and marked most deprecated. does cobra just pick all the global ones up again?

Reply:
> this was selectively adding specific global flags, not all global flags, and marked most deprecated. does cobra just pick all the global ones up again?

It is consistent with the other commands.  I'd much prefer to build a common whitelist if its a thing we want.  After this, we have https://github.com/kubernetes/kubernetes/pull/58408 and then we can start dealing with all the command uniformally and fix it everywhere.

Reply:
Ok, I can update this tomorrow morning